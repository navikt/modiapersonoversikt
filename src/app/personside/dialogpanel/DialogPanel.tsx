import * as React from 'react';
import styled from 'styled-components/macro';
import ErrorBoundary from '../../../components/ErrorBoundary';
import { useAppState } from '../../../utils/customHooks';
import SendNyMeldingContainer from './sendMelding/SendNyMeldingContainer';
import FortsettDialogContainer from './fortsettDialog/FortsettDialogContainer';
import useVisTraadTilknyttetPlukketOppgave from './fortsettDialog/useVisTraadTilknyttetPlukketOppgave';

const DialogPanelWrapper = styled.div`
    flex-grow: 1;
    word-break: break-word;
`;

function DialogPanel() {
    const dialogpanelTraad = useAppState(state => state.oppgaver.dialogpanelTraad);
    const slåOppOppgave = useVisTraadTilknyttetPlukketOppgave(dialogpanelTraad);

    if (slåOppOppgave.pending) {
        return slåOppOppgave.placeholder;
    }

    return (
        <ErrorBoundary boundaryName="Dialogpanel">
            <DialogPanelWrapper>
                {dialogpanelTraad ? (
                    <FortsettDialogContainer
                        traad={dialogpanelTraad}
                        key={dialogpanelTraad.traadId} // for å tvinge refresh dersom man velger en ny tråd
                    />
                ) : (
                    <SendNyMeldingContainer />
                )}
            </DialogPanelWrapper>
        </ErrorBoundary>
    );
}

export default React.memo(DialogPanel);
