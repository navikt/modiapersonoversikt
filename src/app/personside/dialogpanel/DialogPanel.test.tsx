import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TestProvider from '../../../test/Testprovider';
import DialogPanel from './DialogPanel';

test('viser dialogpanel', () => {
    const dialogPanelBody = renderer.create(
        <TestProvider>
            <DialogPanel />
        </TestProvider>
    );

    expect(dialogPanelBody.toJSON()).toMatchSnapshot();
});
