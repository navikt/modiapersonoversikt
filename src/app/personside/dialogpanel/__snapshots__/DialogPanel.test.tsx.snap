// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viser dialogpanel 1`] = `
.c12 label {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px 1px 1px 1px);
  clip: rect(1px 1px 1px 1px);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c14 {
  background-color: white;
  border-radius: .25rem;
  border: 1px solid #78706a;
  position: relative;
}

.c15 {
  border: none;
  cursor: pointer;
  background-color: transparent;
  padding: 0.5rem;
  border-radius: .25rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c15:focus {
  outline: none;
  box-shadow: 0 0 0 0.1875rem #254b6d;
}

.c5 {
  position: absolute;
  height: 2rem;
  width: 2rem;
  border-bottom-right-radius: 50%;
  padding: 0;
  background-color: #0067c5;
  border: none;
  z-index: 10;
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.c5:after {
  background: none;
}

.c5:hover {
  -webkit-transform: none;
  -ms-transform: none;
  transform: none;
}

.c6 {
  position: relative;
  top: 2px;
  width: 18px;
}

.c9 {
  position: absolute;
  top: 2.8rem;
  left: -1rem;
}

.c9 ul {
  list-style: circle;
  margin-top: 1rem;
}

.c9 ul li {
  margin-left: 1.5rem;
}

.c8 {
  position: relative;
}

.c10 {
  text-align: right;
  font-style: italic;
  color: #78706a;
}

.c4 {
  position: relative;
}

.c4 textarea {
  min-height: 9rem;
}

.c4 label {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px 1px 1px 1px);
  clip: rect(1px 1px 1px 1px);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c7 textarea,
.c7 .textareamirror {
  padding-left: 2.25rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c3 .skjemaelement {
  margin-bottom: 0;
}

.c3 > * {
  margin-bottom: 1rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c11 > * {
  margin-top: 0.3rem;
}

.c11 > *:not(:last-child) {
  margin-right: 1rem;
}

.c16 {
  margin-top: 1rem;
}

.c16 label {
  position: absolute !important;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px 1px 1px 1px);
  clip: rect(1px 1px 1px 1px);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
}

.c1 {
  padding: 1rem .5rem;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c17 > * {
  margin-bottom: 0.7rem;
}

.c13 {
  margin-top: 1rem;
}

.c2 {
  margin-bottom: 1rem !important;
}

.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  word-break: break-word;
}

<div
  className="c0"
>
  <article
    aria-describedby="Helt tilfeldig ID"
    className="c1"
  >
    <h2
      className="typo-undertittel c2"
      id="Helt tilfeldig ID"
    >
      Send ny melding
    </h2>
    <form
      className="c3"
      onSubmit={[Function]}
    >
      <div
        className="c4"
      >
        <button
          className="knapp c5 knapp--hoved"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <svg
            className="c6"
            height="24px"
            preserveAspectRatio="xMidYMid meet"
            viewBox="0 0 24 24"
            width="24px"
            x="0px"
            xmlns="http://www.w3.org/2000/svg"
            y="0px"
          >
            <path
              d="M20.854,6.146l-6-6C14.76,0.053,14.632,0,14.5,0h-11C3.224,0,3,0.224,3,0.5v23C3,23.776,3.224,24,3.5,24h17  c0.276,0,0.5-0.224,0.5-0.5v-17C21,6.367,20.947,6.24,20.854,6.146z M7.5,7H12c0.276,0,0.5,0.224,0.5,0.5S12.276,8,12,8H7.5  C7.224,8,7,7.776,7,7.5S7.224,7,7.5,7z M16.5,20h-9C7.224,20,7,19.776,7,19.5S7.224,19,7.5,19h9c0.276,0,0.5,0.224,0.5,0.5  S16.776,20,16.5,20z M16.5,17h-9C7.224,17,7,16.776,7,16.5S7.224,16,7.5,16h9c0.276,0,0.5,0.224,0.5,0.5S16.776,17,16.5,17z   M16.5,14h-9C7.224,14,7,13.776,7,13.5S7.224,13,7.5,13h9c0.276,0,0.5,0.224,0.5,0.5S16.776,14,16.5,14z M16.5,11h-9  C7.224,11,7,10.776,7,10.5S7.224,10,7.5,10h9c0.276,0,0.5,0.224,0.5,0.5S16.776,11,16.5,11z M14.5,6.5v-6l6,6H14.5z"
              fill="currentColor"
            />
          </svg>
          <span
            className="sr-only"
          >
            Standardtekster
          </span>
        </button>
        <div
          className="c7"
        >
          <div
            className="c8"
          >
            <div
              className="skjemaelement textarea__container"
            >
              <label
                className="skjemaelement__label"
                htmlFor="Helt tilfeldig ID"
              >
                Melding
              </label>
              <div
                className="textarea--medMeta__wrapper"
              >
                <textarea
                  className="skjemaelement__input textarea--medMeta"
                  id="Helt tilfeldig ID"
                  onChange={[Function]}
                  onKeyDown={[Function]}
                  placeholder="Alt du skriver i denne boksen blir synlig for Aremark når du trykker \\"Del med Aremark\\""
                  style={
                    Object {
                      "height": "30px",
                    }
                  }
                  type="text"
                  value=""
                />
              </div>
              <div
                aria-live="assertive"
                role="alert"
              />
              <div
                aria-hidden="true"
                className="textareamirror"
              />
            </div>
            <div
              className="c9"
            >
              <div
                className="hjelpetekst"
              >
                <button
                  aria-label="Hjelptekst"
                  aria-pressed={false}
                  className="hjelpetekst__apneknapp"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  title="Hjelptekst"
                  type="button"
                >
                  <span
                    className="sr-only"
                  >
                    Hjelptekst
                  </span>
                  <svg
                    className="hjelpetekst__anchor"
                    focusable="false"
                    height={32}
                    kind="help-circle"
                    viewBox="0 0 18.2 18.2"
                    width={32}
                  >
                    <path
                      d="M9.1 0C4.1 0 0 4.1 0 9.1s4.1 9.1 9.1 9.1 9.1-4.1 9.1-9.1S14.1 0 9.1 0zm0 17.2C4.6 17.2 1 13.6 1 9.1S4.6 1 9.1 1s8.1 3.6 8.1 8.1-3.6 8.1-8.1 8.1z"
                    />
                    <circle
                      cx="9.1"
                      cy="13.8"
                      r=".9"
                    />
                    <path
                      d="M9.1 11.5c-.3 0-.5-.2-.5-.5V8.6c0-.3.2-.5.5-.5 1 0 1.9-.8 1.9-1.9s-.8-1.9-1.9-1.9c-1 0-1.9.8-1.9 1.9 0 .3-.2.5-.5.5s-.5-.2-.5-.4c0-1.6 1.3-2.9 2.9-2.9S12 4.7 12 6.3c0 1.4-1 2.6-2.4 2.8V11c0 .3-.2.5-.5.5z"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <p
              className="typo-normal c10"
              feil={false}
            >
              Du har 
              5000
               tegn igjen
            </p>
          </div>
        </div>
        <div
          aria-live="assertive"
          role="alert"
        />
      </div>
      <div
        className="c11"
      >
        <div
          className="skjemaelement skjemaelement--horisontal"
        >
          <input
            checked={true}
            className="skjemaelement__input radioknapp"
            id="Helt tilfeldig ID"
            name="dialogtype"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="skjemaelement__label"
            htmlFor="Helt tilfeldig ID"
          >
            Referat telefon
          </label>
        </div>
        <div
          className="skjemaelement skjemaelement--horisontal"
        >
          <input
            checked={false}
            className="skjemaelement__input radioknapp"
            id="Helt tilfeldig ID"
            name="dialogtype"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="skjemaelement__label"
            htmlFor="Helt tilfeldig ID"
          >
            Referat oppmøte
          </label>
        </div>
        <div
          className="skjemaelement skjemaelement--horisontal"
        >
          <input
            checked={false}
            className="skjemaelement__input radioknapp"
            id="Helt tilfeldig ID"
            name="dialogtype"
            onChange={[Function]}
            type="radio"
          />
          <label
            className="skjemaelement__label"
            htmlFor="Helt tilfeldig ID"
          >
            Spørsmål
          </label>
        </div>
      </div>
      <div
        className=""
      >
        <div
          className=""
        >
          <div
            className="skjemaelement c12"
          >
            <label
              className="skjemaelement__label"
              htmlFor="Helt tilfeldig ID"
            >
              Tema
            </label>
            <div
              className="selectContainer input--fullbredde"
            >
              <select
                className="skjemaelement__input"
                disabled={false}
                id="Helt tilfeldig ID"
                onChange={[Function]}
                value=""
              >
                <option
                  disabled={true}
                  value=""
                >
                  Velg temagruppe
                </option>
                <option
                  value="ARBD"
                >
                  Arbeid
                </option>
                <option
                  value="FMLI"
                >
                  Familie
                </option>
                <option
                  value="HJLPM"
                >
                  Hjelpemidler
                </option>
                <option
                  value="PENS"
                >
                  Pensjon
                </option>
                <option
                  value="OVRG"
                >
                  Øvrige henvendelser
                </option>
              </select>
            </div>
            <div
              aria-live="assertive"
              role="alert"
            />
          </div>
          <div
            aria-live="assertive"
            role="alert"
          />
        </div>
        <div
          className="alertstripe c13 alertstripe--info"
        >
          <span
            aria-label="info"
            className="alertstripe__ikon"
          >
            <svg
              focusable="false"
              height="1.5em"
              kind="info-sirkel-fyll"
              viewBox="0 0 24 24"
              width="1.5em"
            >
              <g
                fill="none"
              >
                <path
                  d="M12 0C5.382 0 0 5.382 0 12s5.382 12 12 12c6.617 0 12-5.382 12-12S18.617 0 12 0z"
                  fill="#669DB4"
                />
                <path
                  d="M12 5a1.566 1.566 0 1 1 .11 3.13A1.566 1.566 0 0 1 12 5zm2.976 12.01c.563 0 1.043.431 1.043.991s-.48.992-1.043.992H9.39c-.564 0-1.043-.431-1.043-.992 0-.56.479-.99 1.043-.99h1.6v-5.016h-.986c-.565 0-1.044-.43-1.044-.991 0-.56.48-.991 1.044-.991h2.03c.563 0 1.043.43 1.043.99v6.007h1.899z"
                  fill="#FFF"
                />
              </g>
            </svg>
          </span>
          <div
            className="typo-normal alertstripe__tekst"
          >
            Gir ikke varsel til bruker
          </div>
        </div>
        <div
          className=""
        >
          <div
            className="c14"
          >
            <button
              aria-expanded={false}
              className="c15"
              onClick={[Function]}
              type="button"
            >
              <p
                className="typo-normal"
              >
                Velg sak
              </p>
              <i
                className="nav-frontend-chevron chevronboks chevron--ned"
              />
            </button>
          </div>
          <div
            aria-live="assertive"
            role="alert"
          />
        </div>
        <div
          className="skjemaelement c16"
        >
          <label
            className="skjemaelement__label"
            htmlFor="Helt tilfeldig ID"
          >
            Oppgaveliste
          </label>
          <div
            className="selectContainer input--fullbredde"
          >
            <select
              className="skjemaelement__input"
              disabled={false}
              id="Helt tilfeldig ID"
              onChange={[Function]}
              value="MinListe"
            >
              <option
                value="MinListe"
              >
                Svar skal til min oppgaveliste
              </option>
              <option
                value="Enhetensliste"
              >
                Svar skal til 
                NAV Testmark
                 sin oppgaveliste
              </option>
            </select>
          </div>
          <div
            aria-live="assertive"
            role="alert"
          />
        </div>
        <div
          className="alertstripe c13 alertstripe--info"
        >
          <span
            aria-label="info"
            className="alertstripe__ikon"
          >
            <svg
              focusable="false"
              height="1.5em"
              kind="info-sirkel-fyll"
              viewBox="0 0 24 24"
              width="1.5em"
            >
              <g
                fill="none"
              >
                <path
                  d="M12 0C5.382 0 0 5.382 0 12s5.382 12 12 12c6.617 0 12-5.382 12-12S18.617 0 12 0z"
                  fill="#669DB4"
                />
                <path
                  d="M12 5a1.566 1.566 0 1 1 .11 3.13A1.566 1.566 0 0 1 12 5zm2.976 12.01c.563 0 1.043.431 1.043.991s-.48.992-1.043.992H9.39c-.564 0-1.043-.431-1.043-.992 0-.56.479-.99 1.043-.99h1.6v-5.016h-.986c-.565 0-1.044-.43-1.044-.991 0-.56.48-.991 1.044-.991h2.03c.563 0 1.043.43 1.043.99v6.007h1.899z"
                  fill="#FFF"
                />
              </g>
            </svg>
          </span>
          <div
            className="typo-normal alertstripe__tekst"
          >
            Gir varsel, bruker kan svare
          </div>
        </div>
      </div>
      <div
        className="c17"
      >
        <button
          className="knapp knapp--hoved"
          disabled={false}
          type="submit"
        >
          Del med 
          Aremark
        </button>
      </div>
    </form>
  </article>
</div>
`;
